File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers) - 1
        
        while left < right:
            numbers_sum = numbers[left] + numbers[right]

            if numbers_sum == target:
                return [left + 1, right + 1]
            elif numbers_sum < target:
                left += 1
            else:
                right -= 1

File: Data Structures & Algorithms/binary-tree-diameter/submission-1.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    private diameter: number = 0

    private descend(node: TreeNode | null): number {
        if (!node) {
            return 0
        }

        const heightLeft = this.descend(node.left)
        const heightRight = this.descend(node.right)

        this.diameter = Math.max(this.diameter, heightLeft + heightRight)

        return Math.max(heightLeft, heightRight) + 1
    }

    /**
     * @param {TreeNode} root
     * @return {number}
     */
    diameterOfBinaryTree(root: TreeNode | null): number {
        this.diameter = 0
        this.descend(root)

        return this.diameter
    }
}
